logfire/_internal/integrations/llm_providers/llm_provider.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, AsyncIterator, Callable, ContextManager, Iterator
+
+from opentelemetry import context
+
+from ...constants import ONE_SECOND_IN_NANOSECONDS
+
+if TYPE_CHECKING:
+    from ...main import Logfire, LogfireSpan
+    from .types import EndpointConfig
+
+
+__all__ = ('instrument_llm_provider',)
+
+
+def instrument_llm_provider(
+    logfire: Logfire,
+    client: Any,
+    suppress_otel: bool,
+    scope_suffix: str,
+    get_endpoint_config_fn: Callable[[Any], EndpointConfig],
+    on_response_fn: Callable[[Any, LogfireSpan], Any],
+    is_async_client_fn: Callable[[Any], bool],
+) -> ContextManager[None]:
+    """Instruments the provided `client` with `logfire`."""
+    logfire_llm = logfire.with_settings(custom_scope_suffix=scope_suffix.lower(), tags=['llm'])
```

Comment:
We need the `llm` tag for things to show up nicely in the UI. Unfortunately `LLM` isn't caught. Options seem like:
1. Update UI to notice `LLM` tag as well (or just not have case sensitivity?).
2. Update UI to enable the UI if the `custom_scope_suffix` is Anthropic, in which case we can remove the `llm` tag from the SDK internals (but leave in the UI for external integrations such as Mirascope).

Reply:
I think this depends on the resolution of https://github.com/pydantic/logfire/pull/181/files/14193d540735896728724e5943459d83d08e01fe#r1603349085, because it seems like openai and anthropic will each have their own code paths in the frontend.

Reply:
For now looks like best solution would be to support both `llm` and `LLM` so that internally only `LLM` is used but integrations (like Mirascope) won't break.